src/components/Roles/RolesPage.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  Button,
+  DataGrid,
+  DataGridBody,
+  DataGridCell,
+  DataGridHeader,
+  DataGridHeaderCell,
+  DataGridProps,
+  DataGridRow,
+  Spinner,
+  TableCellLayout,
+  TableColumnDefinition,
+  TableRowId,
+  createTableColumn,
+} from "@fluentui/react-components";
+import { AddIcon, DeleteIcon } from "@fluentui/react-icons-mdl2";
+import { Role, useDeleteRole, useRoles } from "api/rolesApi";
+import UserAvatar from "components/RequestsTable/UserAvatar";
+import { useState } from "react";
+
+const roleColumns: TableColumnDefinition<Role>[] = [
+  createTableColumn<Role>({
+    columnId: "role",
+    compare: (a, b) => {
+      return a.Title.localeCompare(b.Title);
+    },
+    renderHeaderCell: () => {
+      return "Role";
+    },
+    renderCell: (item) => {
+      return <TableCellLayout>{item.Title}</TableCellLayout>;
+    },
+  }),
+  createTableColumn<Role>({
+    columnId: "user",
+    compare: (a, b) => {
+      return a.user.Title.localeCompare(b.user.Title);
+    },
+    renderHeaderCell: () => {
+      return "User";
+    },
+    renderCell: (item) => {
+      return (
+        <TableCellLayout
+          style={{ whiteSpace: "nowrap", minWidth: "fit-content" }}
+          media={<UserAvatar user={item.user} />}
+        >
+          {item.user.Title}
+        </TableCellLayout>
+      );
+    },
+  }),
+];
+
+const RolesPage = () => {
+  const roles = useRoles();
+  const [selectedRows, setSelectedRows] = useState(new Set<TableRowId>([]));
+  const [_panelOpen, setPanelOpen] = useState(false);
+  const deleteRole = useDeleteRole();
+
+  const onSelectionChange: DataGridProps["onSelectionChange"] = (_e, data) => {
+    setSelectedRows(data.selectedItems);
+  };
+
+  return (
+    <div style={{ width: "fit-content" }}>
+      {roles.isLoading && <Spinner label="Loading..." />}
+      <Button
+        appearance="subtle"
+        icon={<AddIcon />}
+        onClick={() => {
+          setSelectedRows(new Set());
+          setPanelOpen(true);
+        }}
+      >
+        Add User
+      </Button>
+      <Button
+        appearance="subtle"
+        icon={<DeleteIcon />}
+        disabled={selectedRows.size !== 1}
+        onClick={() => {
```

Comment:
Not sure this is right... isn't item a TableRowId --- which won't always match the SharePoint ID

Reply:
You sure can, and do! :)